src/w3c/templates/headers.js

```diff
@@ -89,29 +89,42 @@ const localizationStrings = {
 
 export const l10n = getIntlData(localizationStrings);
 
-function getSpecTitleElem(conf) {
+export function getSpecTitleElem(conf) {
   const specTitleElem =
     document.querySelector("h1#title") || document.createElement("h1");
+
   if (specTitleElem.parentElement) {
     specTitleElem.remove();
+
+    if (specTitleElem.textContent.trim() === "") {
+      pub(
+        "warn",
+        'textContent of h1#title element is "" or whitespace. Falling back to `document.title` or "No Title"'
+      );
+      specTitleElem.textContent = document.title || "No Title";
+    }
+
+    document.title = norm(specTitleElem.textContent);
   } else {
-    specTitleElem.textContent = conf.title;
+    if (document.title) {
+      specTitleElem.textContent = document.title;
+    } else {
+      specTitleElem.textContent = document.title = "No Title";
+    }
     specTitleElem.id = "title";
   }
+
   if (conf.isPreview && conf.prNumber) {
     const prUrl = conf.prUrl || `${conf.github.repoURL}pull/${conf.prNumber}`;
     const { childNodes } = html`
       Preview of PR <a href="${prUrl}">#${conf.prNumber}</a>:
     `;
     specTitleElem.prepend(...childNodes);
+    document.title = `Preview of PR #${conf.prNumber}: ${document.title}`;
```

Comment:
As above... don't set document.title here... only in one place right at the very end.